Gamma_Analysis.py

```diff
@@ -0,0 +1,341 @@
+"""
+This will perform a gamma analysis
+It will collect two spectra:
+M = Measurement Spectrum
+B = Background Spectrum
+"""
+from __future__ import print_function
+from SpectrumFileBase import SpectrumFileBase
+import Isotope_identification as ii
+import SPEFile
+import math
+import numpy as np
+import matplotlib.pyplot as plt
+import pandas as pd
+import os
+
+EFFICIENCY_CAL_COEFFS = [-5.1164, 161.65, -3952.3, 30908]
+
+
+def absolute_efficiency(energy, coeffs=EFFICIENCY_CAL_COEFFS):
+    """
+    Returns absolute efficiencies for a given set of energies, based on a
+    provided efficiency calibration. It takes an energy (in keV) and a set
+    of calibration coefficients.
+    The efficiency is calculated using the equation given below:
+    ln(efficiency) = c3*(E^3) + c2*(E^2) + c1*E + c0*E,
+    where E = ln(energy[keV])/energy[keV]
+    """
+    efficiency = []
+    for i in range(len(energy)):
+        efficiency.append(math.exp(coeffs[3] *
+                          (math.log(energy[i])/energy[i])**3 +
+                          coeffs[2]*(math.log(energy[i])/energy[i])**2 +
+                          coeffs[1]*(math.log(energy[i])/energy[i]) +
+                          coeffs[0]))
+    return efficiency
+
+
+def emission_rate(net_area, efficiency, livetime):
+    """
+    this function returns the emission rate of gammas per second
+    alongside its uncertainty.
+    """
+    emission_rate = [net_area[0]/(efficiency*livetime),
+                     net_area[1]/(efficiency*livetime)]
+    return emission_rate
+
+
+def Isotope_Activity(Isotope, emission_rates, emission_uncertainty):
+    """
+    Isotope_Activity will determine the activity of a given radioactive isotope
+    based on the emission rates given and the isotope properties. It takes an
+    Isotope object and a given set of emission rates and outputs an activity
+    estimate alongside its uncertainty.
+    """
+    Branching_ratio = Isotope.list_sig_g_b_r
+    Activity = []
+    Uncertainty = []
+    for i in range(len(Branching_ratio)):
+        Activity.append(emission_rates[i]/Branching_ratio[i])
+        Uncertainty.append(emission_uncertainty[i]/Branching_ratio[i])
+    Isotope_Activity = np.mean(Activity)
+    Activity_uncertainty = np.mean(Uncertainty)
+    results = [Isotope_Activity, Activity_uncertainty]
+    return results
+
+
+def peak_measurement(M, energy):
+    """
+    Takes in a measured spectra alongside a specific energy and returns the net
+    area and uncertainty for that energy.
+    """
+    energy_axis = M[0]
+    M_counts = M[1]
+
+    # Rough estimate of FWHM.
+    FWHM = 0.05*energy**0.5
+
+    # Peak Gross Area
+
+    start_peak = 0
+    while energy_axis[start_peak] < (energy - FWHM):
+        start_peak += 1
```

Comment:
Loops are slow and ugly. The better way to do this is with `np.nonzero`:
```
start_peak = np.nonzero(energy_axis > energy - FWHM)[0][0]
```
or a little more elegantly,
```
start_peak = np.flatnonzero(energy_axis > energy - FWHM)[0]
```
Look these up in the numpy documentation and understand what they do. Ask me if you are still confused what the `[0]`'s are for.